pkg/ttl/ttlworker/task_manager.go

```diff
@@ -451,14 +451,19 @@ func (m *taskManager) updateHeartBeat(ctx context.Context, se session.Session, n
 		}
 
 		intest.Assert(se.GetSessionVars().Location().String() == now.Location().String())
-		sql, args, err := updateTTLTaskHeartBeatSQL(task.JobID, task.ScanID, now, state)
+		sql, args, err := updateTTLTaskHeartBeatSQL(task.JobID, task.ScanID, now, state, m.id)
 		if err != nil {
 			return err
 		}
 		_, err = se.ExecuteSQL(ctx, sql, args...)
 		if err != nil {
 			return errors.Wrapf(err, "execute sql: %s", sql)
 		}
+
+		if se.GetSessionVars().StmtCtx.AffectedRows() != 1 {
+			return errors.Errorf("fail to update task status, maybe the owner is not myself, affected rows: %d",
```

Comment:
Do we need to print the current taskManager ID?
Could the `AffectedRows() ` be greater than 1?

Reply:
Good suggestion. Done! :+1: